firewood/src/merkle/node.rs

```diff
@@ -310,6 +310,10 @@ impl ExtNode {
         self.1
     }
 
+    pub fn chd_encoded(&self) -> Option<&Vec<u8>> {
+        self.2.as_ref()
```

Comment:
When it comes to eking out every last drop of performance, it's better to return a pointer with an address and a length rather than return a pointer to another pointer. Just like you should never have a reference to an `Option`, you should probably also never have an _immutable_ reference to a `Vec`. 
```suggestion
    pub fn chd_encoded(&self) -> Option<&[u8]> {
        self.2.as_deref()
```

Reply:
This one isn't a nit

Reply:
nit: using unnamed tuples here really hurts from a readability perspective. We only write code once, but read it many times. Please consider naming the elements of `ExtNode` for another PR, probably something like 0->`partial_path`, 1->`disk_address`, and 2->`child_encoded`. Then, all the callers to `ExtNode::path`, `chd` and `chd_encoded` can be unrolled inline.

Reply:
> This one isn't a nit

sorry I thought I pushed the changes, but it didn't go through... It is now. Thanks!